### PR TITLE
Minor vehicle tidying 2

### PR DESCRIPTION
**Pull request by TiviPlus**
## About The Pull Request

Moving some code, using tool procs, nothing special

## Changelog
:cl:
code: Cleaned up scooter/skate code
/:cl:

### CODE REVIEW

File: code/modules/vehicles/speedbike.dm [hunk 1]
```diff
@@ -3,13 +3,11 @@
...
 	layer = LYING_MOB_LAYER
-	var/overlay_state = "cover_blue"
-	var/mutable_appearance/overlay
+	var/cover_iconstate = "cover_blue"
 
 /obj/vehicle/ridden/speedbike/Initialize()
 	. = ..()
-	overlay = mutable_appearance(icon, overlay_state, ABOVE_MOB_LAYER)
-	add_overlay(overlay)
+	add_overlay(image(icon, cover_iconstate, ABOVE_MOB_LAYER))
```
Comment (LemonInTheDark) [inline code/modules/vehicles/speedbike.dm:10]: isn't it better to cache this? or am I braindead

File: code/modules/vehicles/scooter.dm [hunk 2]
```diff
@@ -12,38 +12,34 @@
 	AddElement(/datum/element/ridable, /datum/component/riding/vehicle/scooter)
 
 /obj/vehicle/ridden/scooter/wrench_act(mob/living/user, obj/item/I)
-	..()
+	. = ..()
```
Comment (LemonInTheDark) [inline code/modules/vehicles/scooter.dm:15]: why set . if you override the default return value all the time?

File: code/modules/vehicles/scooter.dm [hunk 3]
```diff
@@ -12,38 +12,34 @@
...
+	for(var/mob/living/buckled_mob as anything in buckled_mobs)
 		if(buckled_mob.num_legs > 0)
 			buckled_mob.pixel_y = 5
 		else
 			buckled_mob.pixel_y = -4
 
-/obj/vehicle/ridden/scooter/buckle_mob(mob/living/M, force = FALSE, check_loc = TRUE)
-	if(!istype(M))
-		return FALSE
-	return ..()
```
Comment (LemonInTheDark) [inline code/modules/vehicles/scooter.dm:40]: do you know why this was here in the first place?

File: code/modules/vehicles/scooter.dm [hunk 4]
```diff
@@ -214,33 +209,35 @@
...
+		return
+	to_chat(user, "<span class='notice'>You begin making handlebars for [src].</span>")
+	if(!I.use_tool(src, user, 25, volume=50, amount=2))
+		return
+	to_chat(user, "<span class='notice'>You add the rods to [src], creating handlebars.</span>")
+	var/obj/vehicle/ridden/scooter/S = new(loc)
+	if(has_buckled_mobs())
+		var/mob/living/carbon/H = buckled_mobs[1]
+		unbuckle_mob(H)
+		S.buckle_mob(H)
```
Comment (LemonInTheDark) [inline code/modules/vehicles/scooter.dm:224]: Could you make these better?

File: code/modules/vehicles/scooter.dm [hunk 5]
```diff
@@ -214,33 +209,35 @@
...
-			unbuckle_mob(H)
-		qdel(src)
+	if(!I.use_tool(src, user, 20, volume=50))
+		return
+	to_chat(user, "<span class='notice'>You deconstruct the wheels on [src].</span>")
+	new /obj/item/stack/sheet/iron(drop_location(), 5)
+	new /obj/item/scooter_frame(drop_location())
+	if(has_buckled_mobs())
+		var/mob/living/carbon/H = buckled_mobs[1]
+		unbuckle_mob(H)
```
Comment (LemonInTheDark) [inline code/modules/vehicles/scooter.dm:239]: ditto here

File: code/modules/vehicles/scooter.dm (hunk 3 above)
Comment (TiviPlus) [reply code/modules/vehicles/scooter.dm:40]: No idea but it's kinda useless

Comment (LemonInTheDark) [reply code/modules/vehicles/scooter.dm:40]: Mood

File: code/modules/vehicles/speedbike.dm (hunk 1 above)
Comment (TiviPlus) [reply code/modules/vehicles/speedbike.dm:10]: Why would we cache it?

File: code/modules/vehicles/scooter.dm (hunk 2 above)
Comment (TiviPlus) [reply code/modules/vehicles/scooter.dm:15]: Sure I guess

File: code/modules/vehicles/scooter.dm (hunk 4 above)
Comment (LemonInTheDark) [reply code/modules/vehicles/scooter.dm]: to be clear, I was referring to var names

Comment (TiviPlus) [reply code/modules/vehicles/scooter.dm]: Done for both https://github.com/tgstation/tgstation/pull/57626/commits/49336ce06b2aec3b226c40126fe39a4bb0f3afac

**Pull request merged**